fl4health/privacy_mechanisms/dpBinomial.py

```diff
@@ -0,0 +1,38 @@
+from typing import List
+
+import torch
+
+
+# TODO why doesn't mechanism result in output type List[int]
+def BinomialMechanism(query_vector: List[int], N: int, p: float, j: int) -> List[int]:
+    """
+    Privatization of query_vector via additive binomial noise.
+
+    Args:
+        query_vector (List[int]): A vector with integer valued components.
+        N (int): The number N of Bernoulli trials which determines the distribution Bin(N, p).
+        p (float): The success probability p which determines the distribution Bin(N, p).
+        j (int): Determines the quantization scale s = 1/j.
+
+    Returns:
+        List[int]: Privatized query vector via the scaler binomial mechanism applied to each component of query_vector
+
+    Reference
+        cpSGD: Communication-efficient and differentially-private distributed SGD
+        https://proceedings.neurips.cc/paper_files/paper/2018/file/21ce689121e39821d07d04faab328370-Paper.pdf
+    """
+
+    dim = len(query_vector)
+    assert dim > 0  # nonempty query_vector
+
+    binomial_samples = torch.distributions.Binomial(dim, p * torch.ones(dim)).sample()
```

Comment:
rather than using the full name here, you could just do
```
from torch.distributions import Binomial
```
as you do a bit later in the PR

Reply:
I was hoping doing this emphasize to the code readers this particular `Binomial` is from torch and not one I defined in this file which is also the "binomial" mechanism. Actually I should do as you suggested but leave a comment to avoid ambiguity.